src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -388,26 +388,25 @@ func (cfg Config) doFetchCloudConfig(proxyAddr string) ([]byte, error) {
 	return ioutil.ReadAll(gzReader)
 }
 
-// updateFrom creates a new Config by merging the given yaml into this Config.
-// Any servers in the updated yaml replace ones in the original Config and any
-// masquerade sets in the updated yaml replace ones in the original Config.
+// updateFrom creates a new Config by 'merging' the given yaml into this Config.
+// The masquerade sets and the collections of servers in the update yaml
+// completely replace the ones in the original Config.
 func (updated *Config) updateFrom(updateBytes []byte) error {
+	// XXX: does this need a mutex, along with everyone that uses the config?
```

Comment:
I gotta say the option @fffw raised of just making the cloud config and user configs separate is appealing to me @aranhoide. We could also do the following that would make testing much simpler:

1. Always have the cloud.yaml reflect the global cloud.yaml
1. Have a separate local-cloud.yaml file that would purely be for testing/development that would completely overwrite cloud.yaml
1. In addition have a separate local-config.yaml file that would exclusively have settings that are only relevant for the client side.

I feel like that makes things significantly simpler conceptually. The names of all of the above files are not at all fixed of course.

What do you think? Could also happen in a future pass, but given that we have almost no user settings now it might be an easy time to change it too.